📝(doc) create complete documentation website for Meet (#1436)
numerique-gouv/meet: docs/reference/env-variables.md

| `DB_PASSWORD` | **Yes** | -- | Database password |

### Redis

| Variable | Required | Default | Description |
|---|---|---|---|
| `REDIS_URL` | **Yes** | -- | Full Redis URL (e.g., `redis://redis:6379/0`) |
| `SESSION_ENGINE` | No | cache backend | Django session storage backend |

### OIDC / Authentication

| Variable | Required | Default | Description |
|---|---|---|---|
| `OIDC_RP_CLIENT_ID` | **Yes** | `meet` | OIDC client ID |
| `OIDC_RP_CLIENT_SECRET` | **Yes** | -- | OIDC client secret |
| `OIDC_RP_SIGN_ALGO` | No | `RS256` | Token signing algorithm |
| `OIDC_RP_SCOPES` | No | `openid email` | Requested OIDC scopes |
| `OIDC_OP_JWKS_ENDPOINT` | **Yes** | -- | OIDC JWKS URL |
| `OIDC_OP_AUTHORIZATION_ENDPOINT` | **Yes** | -- | OIDC authorization URL |
| `OIDC_OP_TOKEN_ENDPOINT` | **Yes** | -- | OIDC token exchange URL |
| `OIDC_OP_USER_ENDPOINT` | **Yes** | -- | OIDC userinfo URL |
| `OIDC_OP_USER_ENDPOINT_FORMAT` | No | `AUTO` | Format of the userinfo response: `AUTO` (detect), `JWT`, or `JSON` |
| `OIDC_OP_LOGOUT_ENDPOINT` | No | -- | OIDC logout URL |
| `OIDC_CREATE_USER` | No | `True` | Automatically create a local user if none exists on first login |

> | `OIDC_FALLBACK_TO_EMAIL_FOR_IDENTIFICATION` | No | `False` | Fall back to matching users by email if `sub` claim doesn't match. Enable only if emails are unique across your provider. |

Reviewer: ```suggestion
| `OIDC_FALLBACK_TO_EMAIL_FOR_IDENTIFICATION` | No | `False` | Fallback to matching users by email if `sub` claim doesn't match. Enable only if emails are unique across your provider. |
```

| `RECORDING_STORAGE_EVENT_ENABLE` | No | `False` | Enable MinIO/S3 bucket event notifications. Required for the backend to be notified when a recording file is uploaded. |
| `RECORDING_ENABLE_STORAGE_EVENT_AUTH` | No | `True` | Require Bearer token for the storage webhook. Set to `False` for the simplest setup. |
| `RECORDING_STORAGE_EVENT_TOKEN` | If auth enabled | -- | Bearer token MinIO sends with storage webhook requests. |
| `RECORDING_EXPIRATION_DAYS` | No | -- | Days before recordings expire. `null` means no expiration. |
| `RECORDING_MAX_DURATION` | No | -- | Maximum recording duration in milliseconds. `null` means no limit. |
| `RECORDING_ENCODING_ENABLED` | No | `False` | Enable custom encoding for recordings. When `False`, LiveKit uses its built-in H264_720P_30 preset. |
| `RECORDING_ENCODING_WIDTH` | No | `1280` | Recording width in pixels |
| `RECORDING_ENCODING_HEIGHT` | No | `720` | Recording height in pixels |
| `RECORDING_ENCODING_FRAMERATE` | No | `30` | Recording frame rate |
| `RECORDING_ENCODING_VIDEO_BITRATE_KBPS` | No | `3000` | Video bitrate in kbps |
| `RECORDING_ENCODING_AUDIO_BITRATE_KBPS` | No | `128` | Audio bitrate in kbps |

> | `RECORDING_ENCODING_KEY_FRAME_INTERVAL_S` | No | `4.0` | Key frame interval in seconds |

Reviewer: @cameledev we will update these with your recent changes.

### Object Storage (S3 / MinIO)

> Required for recording functionality.

Reviewer: it also required for the custom background functionality if file are persisted

| `AWS_S3_SECURE_ACCESS` | No | `True` | Used in egress configuration (`livekit-egress.yaml`). Set to `False` for local MinIO over HTTP. Note: this is an egress-level setting, not a native Django backend setting. |

### Email

| Variable | Required | Default | Description |
|---|---|---|---|
| `DJANGO_EMAIL_HOST` | No | `localhost` | SMTP server hostname |
| `DJANGO_EMAIL_PORT` | No | `25` | SMTP port |
| `DJANGO_EMAIL_HOST_USER` | No | -- | SMTP username |
| `DJANGO_EMAIL_HOST_PASSWORD` | No | -- | SMTP password |
| `DJANGO_EMAIL_USE_TLS` | No | `False` | Enable STARTTLS |
| `DJANGO_EMAIL_USE_SSL` | No | `False` | Enable SSL (mutually exclusive with TLS) |
| `DJANGO_EMAIL_FROM` | No | -- | From address for outgoing emails |
| `DJANGO_EMAIL_BRAND_NAME` | No | -- | Brand name in email templates |
| `DJANGO_EMAIL_LOGO_IMG` | No | -- | Logo URL for email templates |
| `DJANGO_EMAIL_SUPPORT_EMAIL` | No | -- | Support contact address shown in email templates |
| `DJANGO_EMAIL_DOMAIN` | No | -- | Domain shown in email templates |
| `DJANGO_EMAIL_APP_BASE_URL` | No | -- | Base URL of the Meet frontend (used in email links) |

> ### Frontend feature flags

Reviewer: ```suggestion
### Frontend Configurations
```

| Variable | Required | Default | Description |
|---|---|---|---|
| `FRONTEND_CUSTOM_CSS_URL` | No | -- | URL (or path) to a custom CSS file loaded at runtime. Use this to override CSS variables for colors, fonts, and spacing without rebuilding the image. Example: `https://your-domain.com/custom.css` or `/custom.css` (relative to the Meet domain). |
| `FRONTEND_SILENCE_LIVEKIT_DEBUG` | No | `False` | Suppress LiveKit debug logs |

> | `FRONTEND_IS_SILENT_LOGIN_ENABLED` | No | `True` | Enable silent OIDC token refresh. Requires your provider to support `prompt=none`. |

Reviewer: ```suggestion
| `FRONTEND_IS_SILENT_LOGIN_ENABLED` | No | `True` | Enable silent OIDC authentification. Requires your provider to support `prompt=none`. |
```

| `FRONTEND_IDLE_DISCONNECT_WARNING_DELAY` | No | -- | Seconds before idle disconnect warning is shown. When unset, idle disconnect is disabled. |
| `FRONTEND_EXTERNAL_HOME_URL` | No | -- | Custom URL for the home button |
| `FRONTEND_USE_PROCONNECT_BUTTON` | No | `False` | Show ProConnect login button (French gov) |
| `FRONTEND_USE_FRENCH_GOV_FOOTER` | No | `False` | Show French government footer |

> | `FILE_UPLOAD_ENABLED` | No | `False` | Enable custom background image uploads (max 10 per user, 2 MB, JPEG/PNG only). Set `FILE_UPLOAD_RESTRICTIONS` to override limits. |

Reviewer: I would probably create a section for the custom background feature. @FloChehab wdyt?

### Lobby / Waiting room

These control the behavior of the waiting room (lobby). All are optional - the defaults work for most deployments.

| Variable | Required | Default | Description |
|---|---|---|---|
| `LOBBY_WAITING_TIMEOUT` | No | `3` | Seconds a participant's entry request stays in the waiting state before expiring |
| `LOBBY_DENIED_TIMEOUT` | No | `5` | Seconds before a denied participant may re-request entry |
| `LOBBY_ACCEPTED_TIMEOUT` | No | `21600` | Seconds (6 hours) that an accepted lobby token remains valid. Controls how long after admission a participant can actually join. |
| `LOBBY_KEY_PREFIX` | No | `room_lobby` | Redis key prefix for lobby state |
| `LOBBY_NOTIFICATION_TYPE` | No | `participantWaiting` | LiveKit data channel notification type sent to the room when someone is waiting |
| `LOBBY_COOKIE_NAME` | No | `lobbyParticipantId` | Cookie name used to track anonymous lobby participants |

### CORS

> Required when embedding Meet in iframes or calling the API from a different origin (e.g., external integrations or add-ins).

Reviewer: we don't use these settings for external integration or add-in in our deployment.
If it's rendered in an iframe, I guess the requests still originate from the meet domain.

those are just django default settings for security

| `DJANGO_CORS_ALLOWED_ORIGINS` | No | `[]` | List of origins allowed to make cross-origin requests. Example: `["https://app.example.com"]` |
| `DJANGO_CORS_ALLOWED_ORIGIN_REGEXES` | No | `[]` | List of regex patterns matching allowed origins |

### Celery

| Variable | Required | Default | Description |
|---|---|---|---|
| `CELERY_ENABLED` | No | `False` | Enable Celery for async task processing. Currently used for async file deletion cleanup when `FILE_UPLOAD_ENABLED=True`. Recording email notifications are sent synchronously and do not require Celery. |
| `CELERY_BROKER_URL` | No | `redis://redis:6379/0` | Redis URL used as Celery message broker. Defaults to the same Redis instance as the session cache. |

### Summary Service Connection

These variables configure the backend's connection to the summary service (for transcription).

| Variable | Required | Default | Description |
|---|---|---|---|
| `SUMMARY_SERVICE_ENDPOINT` | For transcription | -- | URL of the summary service API endpoint (e.g., `http://summary:8000/api/v1/tasks/`) |

> | `SUMMARY_SERVICE_API_TOKEN` | For transcription | -- | API token for authenticating with the summary service. Must match the `api_key` of the corresponding tenant in the summary service's `AUTHORIZED_TENANTS`. |

Reviewer: @FloChehab out dated now

### Telephony

> 

Reviewer: explain this requires LiveKit sip and a sip trunk provider
